### ISSUE TITLE
JIRA Notification isn't being sent out after job completion

### ISSUE THREAD

**Issue by rhughes1** [rundeck/rundeck#4354]
**Describe the bug**
The job is set to send out a notification to JIRA after the job **successfully** completes using the officially supported JIRA Notification RunDeck plugin found [here](https://github.com/rundeck-plugins/jira-notification). However, looking at the JIRA issue inside JIRA I don't actually see any comment posted by RunDeck.

I was able to confirm using the [JIRA Workflow Step](https://github.com/rundeck-plugins/jira-notification) supported by RunDeck that the issue does indeed exist in my JIRA instance and can successfully perform the request to get the information about the ticket.

**My Rundeck detail**

* Rundeck version: 3.0.11
* install type: rpm
* OS Name/version: CentOS 7.4
* DB Type/version: RDS (MySQL 5.6)
* JIRA Notification Plugin version: 1.0.0
* Java version: 1.8

**To Reproduce**
1. Install the [JIRA Notification](https://github.com/rundeck-plugins/jira-notification) plugin as per the README instructions. Note that the version I used was the version found on GitHub.
2. Create a new job which sends out a JIRA notification on success to a ticket which is passed in by an option. ie The issue key field should be `${option.jira_ticket}`
3. Run the job, passing in a valid JIRA issue ID

**Expected behavior**
A notification should be sent out from RunDeck to the JIRA issue ID specified in the option. When you view the ticket inside JIRA, there should be a comment outlining the result of the job, the options that were passed in, and a link to the job output.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 71

**Additional context**
Looking inside the `service.log`, I do see the following:

```
2019-01-03 08:43:49.885 ERROR --- [eduler_Worker-1] rundeck.services.NotificationService     : Error sending notification: Notification{eventTrigger='onsuccess', type='JIRA', content='{"issue k
ey":"${option.jira_ticket}"}'}: class java.lang.ClassCastException: java.util.Date cannot be cast to java.sql.Timestamp

java.lang.ClassCastException: java.util.Date cannot be cast to java.sql.Timestamp
        at org.rundeck.plugins.notification.JiraNotification.generateMessage(JiraNotification.java:89)
        at org.rundeck.plugins.notification.JiraNotification.postNotification(JiraNotification.java:63)
        at com.dtolabs.rundeck.plugins.notification.NotificationPlugin$postNotification.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:144)
        at rundeck.services.NotificationService.triggerPlugin(NotificationService.groovy:557)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:352)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:68)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:157)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:193)
        at rundeck.services.NotificationService$_triggerJobNotification_closure9.doCall(NotificationService.groovy:405)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
```

I wonder if this plugin has been broken for some time?

**Comment by rhughes1** [comment]
Still experiencing this issue after upgrading my instance to 3.0.12.

Don't know exactly how to bump the importance of this issue...

**Comment by reinerrdeck** [comment]
Maybe the plugin needs some update trying to use the plugin on 3.0.9 and 3.0.11, I open another issue:

https://github.com/rundeck-plugins/jira-notification/issues/13

**Comment by rhughes1** [comment]
@reinerrdeck I confirmed that the [JIRA Workflow Step](https://github.com/rundeck-plugins/jira-notification) plugin works so it's not about the user credentials.

I actually may have a code fix for the JIRA Notification plugin... Currently testing this out now and will create a pull request for it.

**Comment by reinerrdeck** [comment]
You're right rhuges1, my mistake (i misconfigured the plugin and see the same error that you).

**Comment by rhughes1** [comment]
Waiting on https://github.com/rundeck-plugins/jira-notification/pull/14 to be merged in and a new JAR to be produced.
